deck-core/src/main/kotlin/com/deck/core/entity/impl/DeckPrivateChannel.kt

```diff
@@ -1,24 +1,38 @@
 package com.deck.core.entity.impl
 
+import com.deck.common.entity.RawChannel
 import com.deck.common.util.GenericId
 import com.deck.common.util.Timestamp
+import com.deck.common.util.mapToBuiltin
 import com.deck.core.DeckClient
 import com.deck.core.entity.Channel
 import com.deck.core.entity.misc.ChannelContentType
 import com.deck.core.entity.misc.ChannelType
 import java.util.*
 
-public data class DeckPrivateChannel(
+public class DeckPrivateChannel(
     override val client: DeckClient,
-    override val id: UUID,
-    override val name: String,
-    override val description: String,
-    override val type: ChannelType,
-    override val contentType: ChannelContentType,
-    override val createdAt: Timestamp,
-    override val createdBy: GenericId,
-    override val archivedAt: Timestamp?,
-    override val archivedBy: GenericId?,
-    override val updatedAt: Timestamp?,
-    override val deletedAt: Timestamp?,
-) : Channel
+    public val raw: RawChannel
```

Comment:
Here I take back what I said, we should instead of providing raw access, maintain our constructors with all values as it is, since as stated earlier, the entities should be clean and accessible.

Reply:
Would passing all the values already initialized by the constructor not be a loss of performance? Since the developer may not use all of them

Reply:
Yes, that's a problem when passing entities on the constructor, but I've already found a solution and am already working on it.